### ISSUE TITLE
Browse mode > Navigation: odd experience with adding and removing submenu links

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#48741]
Using 6.2-beta 4, I noticed some strangeness where if I add a custom link or page link block, since I can't select or customize the location of those, I'm able to click on them. From there, if I click the + button, it adds a submenu item. If I click on a properly added page link block, it takes me to that page and I'm unable to add a submenu item directly using the + sign (have to use three dot). This entire experience feels janky. I couldn't figure out at first why a submenu was being created until I realized that it was because I could select the empty page link block in a way I couldn't with the others. Tied to this, if you delete a custom link within a submenu, the submenu as an entity isn't removed and remains: 

https://user-images.githubusercontent.com/26996883/222787887-77a8b3e6-78b8-480c-ba98-4a15745b2b7d.mov

To summarize, three issues:
- If you select a not yet set up page link block and click the +, it adds as a submenu.
- When you click the + while selecting a not setup block, the interface for the inserter is different (no ability to search).
- If you remove a submenu item, the submenu section remains. 

cc @WordPress/gutenberg-design as I'm not sure what's intended here.

**Comment by SaxonF** [comment]
Agree this shouldn't occur as its quite unpredictable. Add block should just add menu items at root level regardless of selection.

**Comment by jasmussen** [comment]
Agreed, this is rough in the general navigation block as well.

**Comment by jameskoster** [comment]
>If you select a not yet set up page link block and click the +, it adds as a submenu.
> When you click the + while selecting a not setup block, the interface for the inserter is different (no ability to search).

These two should be fixed by https://github.com/WordPress/gutenberg/pull/48746.

It's a bit awkward that you can add a 'submenu' but not also add a label for that item without going into the full screen editor. I can't really see a quick solution for that in terms of 6.2, except for removing 'Submenu' as an option in the Inserter in the Navigation panel.

**Comment by Mamaduka** [comment]
The Navigation panel path has been removed from 6.2. Punting this to 6.3.

**Comment by ndiego** [comment]
@annezazu I think this now can be closed due to all the updates to the Navigation panel in the Site Editor.

**Comment by annezazu** [comment]
Agreed. Here's a quick visual of what the experience is like currently where the ability to add is removed as of 16.1 RC1:

https://github.com/WordPress/gutenberg/assets/26996883/44763540-fa75-480f-a317-17186b353fbe
